vault/resource_database_secret_backend_connection.go

```diff
@@ -44,27 +44,47 @@ const (
 var (
 	databaseSecretBackendConnectionBackendFromPathRegex = regexp.MustCompile("^(.+)/config/.+$")
 	databaseSecretBackendConnectionNameFromPathRegex    = regexp.MustCompile("^.+/config/(.+$)")
-	dbBackendTypes                                      = []string{
-		dbBackendCassandra,
-		dbBackendCouchbase,
-		dbBackendElasticSearch,
-		dbBackendHana,
-		dbBackendInfluxDB,
-		dbBackendMSSQL,
-		dbBackendMongoDB,
-		dbBackendMongoDBAtlas,
-		dbBackendMySQL,
-		dbBackendMySQLAurora,
-		dbBackendMySQLLegacy,
-		dbBackendMySQLRDS,
-		dbBackendPostgres,
-		dbBackendOracle,
-		dbBackendSnowflake,
-		dbBackendRedshift,
+
+	// map all dbBackends to their corresponding dbEngine's
+	// XXX: This map must be updated whenever a new database engine is added.
+	dbEngines = map[string]*dbEngine{
```

Comment:
Minor struct nit: Does this have to be a map, or can it be `[]*dbEngine{}`? The range loops where this is referenced would still work by changing the map key `k` to object's name `e.name`.

For instance:
```
for _, e := range dbEngines {
	dbBackendTypes = append(dbBackendTypes, e.name)
}
```

Reply:
I believe we still need the map in order to get the `dbEngine` from the Schema's field name , for all callers of `getDBEngine()`

Reply:
I suppose we could iterate through `[]*dbEgine{}` instead of relying on the map.

Reply:
refactored in https://github.com/hashicorp/terraform-provider-vault/pull/1326